cpp_api/api.h

```diff
@@ -0,0 +1,40 @@
+#include <memory>
+#include <vector>
+
+// Forward declaration of API dependencies
+class goto_modelt;
+class message_handlert;
+class optionst;
+
+#include "options.h"
+
+// An object in the pattern of Session Facade - owning all of the memory
+// the API is using and being responsible for the management of that.
+struct cbmc_api
+{
+  // Initialising constructor
+  explicit cbmc_api();
+  cbmc_api(const api_optionst &options);
```

Comment:
🚫 The coding standard requires that constructors which take a single argument are marked `explicit`. Note that this does not apply to constructors with no parameters or with two or more parameters.